trace-utils/src/trace_utils.rs

```diff
@@ -265,10 +265,16 @@ pub fn set_serverless_root_span_tags(span: &mut pb::Span) {
     span.r#type = "serverless".to_string();
     span.meta
         .insert("_dd.origin".to_string(), "gcp_function".to_string());
-    span.meta.insert(
-        "functionname".to_string(),
-        env::var("K_SERVICE").unwrap_or_default(),
-    );
+
+    // Google cloud functions automatically sets either K_SERVICE or FUNCTION_NAME
+    // env vars to denote the cloud function name.
+    // K_SERVICE is set on newer runtimes, while FUNCTION_NAME is set on older deprecated runtimes.
+    if let Ok(function_name) = env::var("K_SERVICE") {
```

Comment:
Same as comment above, its better to rely on a config structure, and more generic way to handle these special cases. Than to add them in the middle of the implemetnation. e.g. If I wanted to use this code in another scenario (outside of GCP) - where someone decided to set the K_SERVICE env - I would get unexpeted span metadata.

Reply:
moved function_name to a config struct